models/migrations/v1_21/v276.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_21 //nolint
+
+import (
+	"code.gitea.io/gitea/models/migrations/base"
+
+	"xorm.io/xorm"
+)
+
+type BadgeUnique struct {
```

Comment:
?
Why that weird name?

Reply:
Because we need two structs, one without the unique constraint and one with, since we can't add the new column with a unique constraint when all the values would be the same (empty). So I add the column, fill it with unique values, then set the constraint.